### PR TITLE
common: add register containing git revision

### PR DESCRIPTION
**Pull request by tjurtsch**
This will help us identify gateware version uploaded to the device.

### CODE REVIEW

**Comment by tjurtsch** [comment]
This PR also adds a helper script that can be used to check git revision stored in the register.

File: rowhammer_tester/scripts/version.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env python3
+
+from rowhammer_tester.scripts.utils import RemoteClient
+
+if __name__ == "__main__":
+    wb = RemoteClient()
+    wb.open()
+
+    # Dump CSR register containing git revision
+    for name, reg in wb.regs.__dict__.items():
```
Comment (jedrzejboczar) [inline rowhammer_tester/scripts/version.py:10]: We could simplify this part to just `wb.regs.gitrev_hash.read()`.

**Review by jedrzejboczar — CHANGES_REQUESTED** [review]
We may also add a CSR with build date and time (up to minutes?). This should not take much space but could be helpful to distinguish builds on the same git revision. Then we could rename `GitRevision` to `BuildInfo` or something similar and have both CSRs created in that module. And if we need more info in the future we would just add CSRs there. And the `version.py` script can print the information (would probably need to convert the date to something readable by humans).

One more thing is that it would be good to add the build info as the first CSR to make sure that it always has the same address even if other CSR locations change. But I am not sure if it is possible to force it's number. We cannot add it before https://github.com/antmicro/litex-rowhammer-tester/blob/master/rowhammer_tester/targets/common.py#L106 because then `self.add_csr` won't work. But after that line all the internal CSRs like `uart_*` are already added. Maybe `csr_id` argument could help (https://github.com/enjoy-digital/litex/blob/master/litex/soc/integration/soc_core.py#L198). It then calls [SoCCSRHandler](https://github.com/enjoy-digital/litex/blob/master/litex/soc/integration/soc.py#L481)'s `add` derived from `SoCLocHandler`. You could check if it is possible and if not we will just have to leave it as it is now.

**Review by tmichalak — APPROVED** [review]
LGTM

File: rowhammer_tester/scripts/version.py (hunk 1 above)
Comment (tjurtsch) [reply rowhammer_tester/scripts/version.py]: Applied

**Review by jedrzejboczar — APPROVED** [review]

**Comment by tjurtsch** [comment]
We moved the BuildInfo CSR at the end of CSR list (we're limited to 32 CSR's). This way it's address should always be the same.

**Pull request merged**